### ISSUE TITLE
Improve web client invite to room UI

### ISSUE THREAD

**Issue by bizzbyster** [RocketChat/feature-requests#60]
In my opinion, the user experience when inviting users to a private room is not very nice. First, even though I typically add the same set of 10 users or so to rooms over and over, the autofill takes forever to figure out who I am trying to add. Also the button says "+add users" after I have already chosen users to add

![image](https://user-images.githubusercontent.com/4001792/54616637-50744f80-4a36-11e9-8ab9-a323aa375e33.png)

But my sense about "+add users" is that it should pop up a new dialog. I'm not sure why I have that sense but in general this user interface could be improved.

**Comment by sudipt1999** [comment]
I will try to give it a shot!

**Comment by bizzbyster** [comment]
"autofill takes forever" -- I noticed that this is not a problem on Android or IOS clients so you may want to compare the way the search is done in those clients and try to replicated it in the web client.

**Comment by sudipt1999** [comment]
"autofill takes forever" its not slow in web also I created 10 users if different names (like sudo1, sudo2 , doodle1, doodle2, test1, test2 ...), So I can confirm that it dosen't takes long, 
>> also we can add same user again and again, that can be fixed by checking if there are user with same username in member list or not
>>About a dialog box ! I can that would be great to be added to a prooduct

**Comment by bizzbyster** [comment]
@sudipt1999 have you tried using a private group in open.rocket.chat web client?

**Comment by sudipt1999** [comment]
Yes @bizzbyster  I have created two channels one public and private to test them out!

**Comment by bhardwajaditya** [comment]
Current search works on pattern matching so may be that's the reason for this long search time. This will occur if there are many usernames with same prefix.

**Comment by sudipt1999** [comment]
@bhardwajaditya  I guess you are right I only had 10 users to test out this thats why it wasn't slow I guess. Will add more users to test it out then.

**Comment by bizzbyster** [comment]
What if there is an optimization to first search for users with which the logged in user has an open DM. Thoughts? @sudipt1999 @bhardwajaditya @tusharjindal353?

**Comment by tusharjindal353** [comment]
@bizzbyster I think we first show the users which are in a conversation with logged in user.Because the probability for adding a user in a group is way more higher for them.What are your thoughts??

**Comment by sudipt1999** [comment]
@tusharjindal353  I agree with u on this, since there is fairly high chance of adding someone who is online first, or with whom they are having conversation so I guess we can just show the logged in members in the search and a button at end to show the users which are logged off currently.

**Comment by tusharjindal353** [comment]
@sudipt1999 @bizzbyster I agreed to u. I think we should first show the member who are already in conversation with logged in user.and in that list online users are kept to be on top of list.So that a admin of a channel can get more relevant and filtered result and we can optimise the things in more better way.What do u say??
